includes/fbs.availability.inc

```diff
@@ -154,8 +166,177 @@ function fbs_availability_holdings($provider_ids) {
 
+  $result = _fbs_set_table_html($holdings, $reservation_count);
   return $result;
+}
 
+/**
+ * Set holdings if material is periodical only.
+ */
+function _fbs_set_holdings_periodical($res) {
+  $holdings = array();
+  $reservation_count = 0;
+  foreach ($res as $id => $records) {
+    $reservation_count += (int) $records['reservation_count'];
+    if (isset($records['holdings'])) {
+      foreach ($records['holdings'] as $holding => $issue_year) {
+        foreach ($issue_year as $key) {
+          $holdings[] = $key[0];
+        }
+      }
+    }
+  }
+
+  $result = _fbs_set_table_html($holdings, $reservation_count);
+  return $result;
+}
+
+/**
+ * Make the html-table.
+ */
+function _fbs_set_table_html($h, $r_count) {
+  // Set a classname for styling the table.
+  $variables['attributes'] = array(
+    'class' => array(
+      drupal_html_class('availability_holdings_table'),
+    ),
+  );
+
+  // Set table header.
+  $variables['header'] = array(
+    'placement' => t('Placement'),
+    'copies' => t('Copies'),
+    'home' => t('At home'),
+  );
+
+  // Set table rows.
+  $variables['rows'] = _fbs_set_rows($h);
+
+  // Set last row with reservation count.
+  $row = array();
+  $row['data']['Library'] = t('Reservations');
+  $row['data']['Copies'] = array('data' => $r_count, 'colspan' => 4);
+  $variables['rows'][] = $row;
+
+  return theme_table($variables);
+}
+
+/**
+ * Set rows in table for given holdings.
+ */
+function _fbs_set_rows($h) {
+  $rows = array();
+
+  $copies_total = 0;
+  $home_total = 0;
+
+  foreach ($h as $key => $data) {
+    $row = array();
+
+    $row['placement'] = implode(' > ', $data['placement']);
+
+    $row['copies'] = isset($data['total_count']) ? (int) $data['total_count'] : 0;
+    $copies_total += $row['copies'];
+
+    $row['home'] = isset($data['available_count']) ? (int) $data['available_count'] : 0;
+    $home_total += $row['home'];
+
+    $rows[] = $row;
+  }
+
+  if (count($rows) >= 1) {
+    $rows = _clean_up_rows($rows);
+  }
+  // Adding row - totals.
+  $row = array();
+  $row['data']['Library'] = t('Total');
+  $row['data']['Copies'] = $copies_total;
+  $row['data']['Home'] = $home_total;
+  $row['class'] = array(drupal_html_class('availability_holdings_last_row'));
+  $rows[] = $row;
+
+  return $rows;
+}
+
+/**
+ * If the same placement exists several times collect them in one line.
+ */
+function _clean_up_rows($_rows) {
+  $rows = array();
+  $placements = array();
+
+  foreach ($_rows as $row) {
+    $currkey = $row['placement'];
+    if (!in_array($currkey, $placements)) {
+      $placements[] = $currkey;
+      $placementsarr = _get_placements_with_key($_rows, $currkey);
+      $this_row = _sum_placement($placementsarr);
+      $rows[] = $this_row;
+    }
+  }
+  return $rows;
+}
+
+/**
+ * Collect materials with the same placement.
+ */
+function _get_placements_with_key($_rows, $currkey) {
+  $rows = array();
+  foreach ($_rows as $key) {
+    if ($key['placement'] == $currkey) {
```

Comment:
you already looping needed placements, why you dont made actions from _sum_placement here?